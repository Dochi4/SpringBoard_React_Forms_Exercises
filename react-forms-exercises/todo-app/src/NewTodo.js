import React, { useState } from "react";

const NewTodo = ({ addTodo }) => {
  const initialState = {
    item: "",
  };

  const [formData, setFormData] = useState(initialState);

  const handleChange = (e) => {
    const { name, value } = e.target;
    setFormData((formData) => ({
      ...formData,
      [name]: value,
    }));
  };

  const handleSubmit = (e) => {
    e.preventDefault();
    if (!formData.item.trim()) return;
    addTodo({
      ...formData,
    });
    setFormData(initialState);
  };

  return (
    <form onSubmit={handleSubmit}>
      <label htmlFor="item">Item</label>
      <input
        type="text"
        id="item"
        name="item"
        value={formData.item}
        onChange={handleChange}
        placeholder="Add Todo Item"
      />
      <button>Add Todo</button>
    </form>
  );
};

export default NewTodo;
