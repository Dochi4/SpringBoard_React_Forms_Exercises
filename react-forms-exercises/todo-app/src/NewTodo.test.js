import React from "react";
import { render, screen, fireEvent } from "@testing-library/react";
import TEST_TODO from "./_testCommon";
import NewTodo from "./NewTodo";

it("should render without crashing", function () {
  render(<NewTodo addTodo={TEST_TODO} />);
});

it("matches snapshot", function () {
  const { asFragment } = render(<NewTodo addTodo={TEST_TODO} />);
  expect(asFragment()).toMatchSnapshot();
});

it("should change the width input", function () {
  render(<NewTodo addTodo={TEST_TODO} />);

  const itemInput = screen.getByLabelText(/item/i);
  fireEvent.change(itemInput, { target: { value: "Newitem" } });
  expect(itemInput.value).toBe("Newitem");
});
