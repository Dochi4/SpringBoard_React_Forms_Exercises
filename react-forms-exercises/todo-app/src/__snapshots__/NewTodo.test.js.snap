// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <form>
    <label
      for="item"
    >
      Item
    </label>
    <input
      id="item"
      name="item"
      placeholder="Add Todo Item"
      type="text"
      value=""
    />
    <button>
      Add Todo
    </button>
  </form>
</DocumentFragment>
`;
